### ISSUE TITLE
External sensor + thermostat capability

### ISSUE THREAD

**Issue by bhaap** [Tomas-Mrazek/SmartThings#3]
After about a week or so the externa sensor stopped updating for me. 

The sensor seem to be working thou (tested by moving it to the fridge, and thus the thermostat opened its valve, and moving it out of the fridge again and after the thermostat closes the valve.)

I do now know what info you would need of me for this.

Also is there any work being done to give this DH thermostat capability? Could really us it for the new Home-assistant integration im using.

**Comment by Tomas-Mrazek** [comment]
Sorry for late reply, I haven't received email notification.

Thermostat capability is no longer officially supported by SmartThings, see [docs](https://smartthings.developer.samsung.com/develop/api-ref/capabilities.html). The issue is with old SmartApss. Google and Alexa integrations are really old and do not work properly.

As for external sensor, try unpair and pair it again. Chances are you're using firmware 4.0. There is a known bug about [external sensor crashing](https://forum.fibaro.com/topic/31373-why-does-44-still-not-work-after-almost-a-year/?do=findComment&comment=161338)

**Comment by carp8000** [comment]
I also installed the new firmware 4.5 for the thermostat. I had both 4.4 and 4.3 - but both seem to lack update of external sensor. Any known issues with external sensor not updating/showing correct reading? I also have Fibaro just for updating Fibaro devices.

**Comment by Tomas-Mrazek** [comment]
Luckily I also have this bug on my devices. :smirk: That must be the reason, why the battery lasted for so long, because polling wasn't working.

I'm not sure, what exactly triggers it, but it seems, that cron job, which updates internal status in SmartThings platform didn't run and thus locked every update, refresh and polling.

I increased time window for automatic status reset, but that might be not enough. As a temporary solution I added a tile named **reset status**. Please, tap it and then manually refresh values (via **refresh** tile). Do this for every locked device and let me know, if that helps.

**Comment by carp8000** [comment]
Got my 4 devices updated to fw 4.5 - but now it’s 4 hours ago and temp. sensor hasn’t changed oven though I used the refresh button - nothing changed.

Den 26. mar. 2019 kl. 16.52 skrev Tomáš Mrázek <notifications@github.com<mailto:notifications@github.com>>:


Luckily I also have this bug on my devices. 😏 That must be the reason, why the battery lasted for so long, because polling wasn't working.

I'm not sure, what exactly triggers it, but it seems, that cron job, which updates internal status in SmartThings platform didn't run and thus locked every update, refresh and polling.

I increased time window for automatic status reset, but that might be not enough. As a temporary solution I added a tile named reset status. Please, tap it and then manually refresh values (via refresh tile). Do this for every locked device and let me know, if that helps.

**Comment by Tomas-Mrazek** [comment]
Have you updated device handler with new source code? And you need to click on new tile called "status reset"

**Comment by carp8000** [comment]
These are the tiles that are present - I did update the Dth.

[image1.png]


Den 26. mar. 2019 kl. 20.30 skrev Tomáš Mrázek <notifications@github.com<mailto:notifications@github.com>>:


Have you updated device handler with new source code? And you need to click on new tile called "status reset"

**Comment by carp8000** [comment]
[image1.jpeg]



Den 26. mar. 2019 kl. 20.30 skrev Tomáš Mrázek <notifications@github.com<mailto:notifications@github.com>>:


Have you updated device handler with new source code? And you need to click on new tile called "status reset"

**Comment by carp8000** [comment]
I now got a new temperature reading 😊

But how often is it supposed to reset status - anyway I can adjust it?

Thank you for the great work!


Den 26. mar. 2019 kl. 20.30 skrev Tomáš Mrázek <notifications@github.com<mailto:notifications@github.com>>:


Have you updated device handler with new source code? And you need to click on new tile called "status reset"

**Comment by Tomas-Mrazek** [comment]
Honestly I cannot say how often. That depends on SmartThings platform reliability. But I'll come with a more bulletproof solution in few days. I'll leave this issue open until I fully resolve it.

**Comment by bhaap** [comment]
Any update to this?

**Comment by Tomas-Mrazek** [comment]
Not yet, my hub still has firmware v25.26 with Z-Wave device bugs that prevent proper pairing. [Here](https://community.smartthings.com/t/hub-firmware-release-notes-25-26/156497/370?u=toka) you can see, what's the problem. It's been almost a month and they still haven't fixed it. I'm slowly thinking about abandoning SmartThings Hub and buy Hubitat instead.

**Comment by bhaap** [comment]
25.32 has now been released. Is this issue fixed now?

**Comment by Tomas-Mrazek** [comment]
No. But I said I need to track the issue, so they created separate thread [here](https://community.smartthings.com/t/known-issue-unable-to-add-fibaro-heat-controller-roller-shutter-smart-implant-qubino-flush-dimmer/164256). Feel free to leave a comment. 

On a side note I got my TRV updated to firmware 4.5, so I'll be able to test algoritm properly next winter.

**Comment by bhaap** [comment]
Here we go again: https://community.smartthings.com/t/hub-firmware-release-notes-26-9/166632

I see "Improvements to pairing process for some Zigbee devices that experienced rare delays or failures"
might this be for fibaro?

**Comment by Tomas-Mrazek** [comment]
Nope, Fibaro TRV are Z-Wave. Guess we still have to wait.

**Comment by Tomas-Mrazek** [comment]
> 
> 
> Any update to this?

Just to let you know @bhaap , the issue with SmartThings platform is now apparently fixed. With the upcoming heating season I will soon update and fix the device handler.

**Comment by bhaap** [comment]
Ah sweet! I just started to use my furnace so this is great news :)

**Comment by Tomas-Mrazek** [comment]
@bhaap There is now official (?) device handler in SmartThings repository. I advise you to use theirs DTH, because they have more information about their own API and source code. I would be fighting against mills. Just remove all related DTH and let SmartThings to automatically select one.

I'm testing it right now and It's okish. It does not support changing properties (LED when remote control, etc), but on the other hand it creates automatically child device for external temperature sensor, which is nice and definitely cleaner solution.

**Comment by carp8000** [comment]
Will try - hopefully the batterydrainage will stop. Within the last 2 weeks I had to recharge the batteries 2 times.

**Comment by Tomas-Mrazek** [comment]
@carp8000 It definitely will stop. The TRVs have it's own temperature range, where they send message to ST Hub. I've read somewhere that the range is as high as 1°C. That means that if your ST shows temperature 22°C, actual temperature can be either 22.9°C or 21.1°C and your ST won't be updated. That's too much bad info for my liking, that's why I implemented custom check initiated from ST. It was configurable, I have set it up to 30 minutes and it was ok. Definitely didn't drain battery that much, I had to recharge it maybe every two months?
